app/src/main/java/com/example/cleanarchitecture/utils/SharedPrefs.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.cleanarchitecture.utils
+
+import android.content.Context
+import android.content.SharedPreferences
+
+@Suppress("UNCHECKED_CAST")
+class SharedPrefs(private val context: Context) {
+    companion object {
+        private const val PREF = "MyAppPrefName"
+        private const val PREF_TOKEN = "user_token"
+    }
+
+    private val sharedPref: SharedPreferences =
+        context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
+
+    fun saveToken(token: String) {
+        put(PREF_TOKEN, token)
+    }
+
+    fun getToken(): String {
+        return get(PREF_TOKEN, String::class.java)
+    }
+
+    private fun <T> get(key: String, clazz: Class<T>): T =
+        when (clazz) {
+            String::class.java -> sharedPref.getString(key, "")
+            Boolean::class.java -> sharedPref.getBoolean(key, false)
+            Float::class.java -> sharedPref.getFloat(key, -1f)
+            Double::class.java -> sharedPref.getFloat(key, -1f)
+            Int::class.java -> sharedPref.getInt(key, -1)
+            Long::class.java -> sharedPref.getLong(key, -1L)
+            else -> null
```

Comment:
лучше будет так 
```suggestion
            else -> error("unsupported type $clazz")
```
если ты захочешь вызвать функцию с типом который у тебя не поддерживается то будет креш
причем потом сложно будет отловить почему тк креш будет на cast exception